learning_resources/etl/canvas.py

```diff
@@ -57,7 +62,9 @@ def run_for_canvas_archive(course_archive_path, overwrite):
     )
     if resource.runs.count() == 0:
         LearningResourceRun.objects.create(
-            run_id=f"{readable_id}+canvas", learning_resource=resource, published=True
+            run_id=f"{readable_id}+canvas",
+            learning_resource=resource,
+            published=True,
```

Comment:
New session-specific courses were created, but the 2 older courses from an earlier command run are still present with `test_mode=True`.  I think any courses not found in the archives during a command run should probably be deleted and/or have test_mode set to `False` (in addition to `published=False`), with all their contentfiles deleted or unpublished from regular and vector search.

Reply:
I added the deletion of stale canvas courses. Retrieving the current course ids required going into the archive so i funnel up the relevant readable ids and at the end of the command do a deletion+deindex. This also only applies when running the management command since the webhook just tells us to add a new entry.